upgrade/1.5.2/Upgrade_NCN_images.md

```diff
@@ -0,0 +1,95 @@
+# Upgrade NCNs during CSM `1.5.2` Patch
+
+This page provides guidance for systems that are performing an upgrade from a CSM `v1.5.X` release to the CSM `v1.5.2` release.
+
+The [`v1.5.2` upgrade page](../1.5.2/README.md) will refer to this page
+during [NCN Upgrade](../1.5.2/README.md#ncn-upgrade).
+
+## Overview
+
+The following steps upgrade NCNs into the node images created during the [Update NCN images](../1.5.2/README.md#update-ncn-images)
+step of the patch procedure. The node upgrades will not change the state of the node.
+
+It is important to do this step so that nodes
+are using the correct images. If NCNs are not upgraded into these images, then `cloudinit` will fail on the nodes the next
+time the nodes are rebuilt. These images were set in BSS during [Update NCN images](../1.5.2/README.md#update-ncn-images)
+and if the NCN nodes are not upgraded, then on any node reboot will cause the node to be booted into this new image which is not following the proper
+upgrade procedure which could cause problems.
+
+## Steps
+
+1. (`ncn-m001#`) Upgrade storage nodes.
+
+    1. Pick one storage node to test the first storage node upgrade on. This will be referred to as the `CANARY_NODE`.
+
+        ```bash
+        CANARY_NODE="ncn-s001"
+        ```
+
+    1. Perform the storage node upgrade on the `CANARY_NODE`.
+
+        ```bash
+        /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ${CANARY_NODE} --upgrade
+        ```
+
+    1. If the `CANARY_NODE` upgrade succeeded, upgrade the remaining storage nodes.
+
+        1. Get a comma sperated list of storage nodes to be upgraded.
+
+            ```bash
+            STORAGE_NODES="$(ceph orch host ls | grep ncn-s | grep -v "$CANARY_NODE" | awk '{print $1}' | tr '\n' ',' | head -c -1)"
+            echo "$STORAGE_NODES"
+            ```
+
+        1. Upgrade remaining storage nodes.
+
+            ```bash
+            /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ${STORAGE_NODES} --upgrade
+            ```
+
+    For troubleshooting the storage node upgrades, see the notes in the [CSM storage node upgrade procedure](../Stage_2.md#storage-node-image-upgrade-and-ceph-upgrade).
+
+1. (`ncn-m001#`) Export `CSM_ARTI_DIR` environment variable. (`CSM_RELEASE_VERSION` and `CSM_DISTDIR` is expected to already be set).
+
+    ```bash
+    CSM_REL_NAME="csm-${CSM_RELEASE_VERSION}"
+    export CSM_ARTI_DIR="${CSM_DISTDIR}"
+    echo "${CSM_ARTI_DIR}"
+    ```
+
+1. Upgrade master nodes and worker nodes.
```

Comment:
@nrockershousen @rustydb @jnowicki-hpe Can you please look over this step? I have changed it and broken out the upgrade of master nodes, worker nodes, and ncn-m001 into different steps. This was done to add instructions to set the environment variables on ncn-m002 before upgrading ncn-m001. This is related to [CASMTRIAGE-7124](https://jira-pro.it.hpe.com:8443/browse/CASMTRIAGE-7124)

Reply:
Please include CASMTRIAGE-7124 in this commit's message.

Reply:
I think this is fine. Aside from my other inline feedback, just amend the relevant commit message mentioning CASMTRIAGE-7124.

Reply:
I fixed the commit message, it now reads "CASMTRIAGE-7106 update 1.5.2 patch, set env variables on m002. This addresses CASMTRIAGE-7124"